src/detect-filesize.c

```diff
@@ -305,26 +289,19 @@ static int DetectFilesizeSetpTest01(void)
     DetectEngineCtx *de_ctx = NULL;
 
     res = DetectFilesizeInitTest(&de_ctx, &sig, &fsd, "1 <> 3 ");
-    if (res == 0) {
-        goto end;
-    }
-
-    if(fsd == NULL)
-        goto cleanup;
+    FAIL_IF(res == 0);
 
-    if (fsd != NULL) {
-        if (fsd->arg1 == 1 && fsd->arg2 == 3 && fsd->mode == DETECT_UINT_RA)
-            res = 1;
-    }
+    FAIL_IF_NULL(fsd);
+    FAIL_IF_NOT(fsd->arg1 == 1);
+    FAIL_IF_NOT(fsd->arg2 == 3);
+    FAIL_IF_NOT(fsd->mode == DETECT_UINT_RA);
 
-cleanup:
-    if (fsd)
-        DetectFilesizeFree(NULL, fsd);
+    DetectFilesizeFree(NULL, fsd);
     SigGroupCleanup(de_ctx);
```

Comment:
just `DetectEngineCtxFree` is enough here, so `SigGroupCleanup` and `SigCleanSignatures` can be removed